storage/src/main/java/tech/pegasys/teku/storage/server/network/DatabaseNetwork.java

```diff
@@ -84,6 +87,15 @@ public static DatabaseNetwork init(
             formatMessage(
                 "deposit contract", depositContractString, databaseNetwork.depositContract));
       }
+      if (!depositContractString.equals(EPHEMERY_DEPOSIT_CONTRACT_ADDRESS)
+          && databaseNetwork.depositChainId != null
+          && !databaseNetwork.depositChainId.equals(depositChainId)) {
```

Comment:
so ideally check not null then check everything else...

Reply:
fixed